File: tcp-serve-write-single.go
package main

import "net"
import "fmt"

func main() {

    var BUF_SIZE = 64 * 1024;
    var NUM_WRITES = 64 * 1024;

    var buf = make([]byte, BUF_SIZE);

    for i := 0; i < BUF_SIZE; i++ {
        buf[i] = (byte)(i % 64 + 48)
    }

    ln, _ := net.Listen("tcp", "localhost:9001")

    conn, _ := ln.Accept()

    for i := 0; i < NUM_WRITES; i++ {
        conn.Write(buf);
    }

    fmt.Println("wrote %u bytes", NUM_WRITES * BUF_SIZE)
}

File: tcp-serve-write-single.rs
use core::rt::io::*;
use core::rt::io::net::tcp::TcpListener;
use core::rt::io::net::ip::Ipv4;

macro_rules! rtdebug (
    ($( $arg:expr),+) => ( {
        dumb_println(fmt!( $($arg),+ ));

        fn dumb_println(s: &str) {
            use core::io::WriterUtil;
            let dbg = ::libc::STDERR_FILENO as ::io::fd_t;
            dbg.write_str(s);
            dbg.write_str("\n");
        }

    } )
)

static BUF_SIZE: uint = 64 * 1024;
static NUM_WRITES: uint = 64 * 1024;

#[start]
fn start(_argc: int, _argv: **u8, _crate_map: *u8) -> int {
    do core::rt::start(argc, argv, crate_map) {
        let addr = Ipv4(127, 0, 0, 1, 9001);
        let mut listener = TcpListener::bind(addr).unwrap();
        let mut stream = listener.accept().unwrap();
        let buf = vec::from_fn(BUF_SIZE, |i| (i % 64 + 48) as u8);
        for NUM_WRITES.times {
            stream.write(buf);
        }
        rtdebug!("wrote %u bytes", buf.len() * NUM_WRITES);
    }
}
